evennia/utils/funcparser.py

```diff
@@ -1252,6 +1252,76 @@ def funcparser_callable_you_capitalize(
     )
 
 
+def funcparser_callable_your(
+    *args, caller=None, receiver=None, mapping=None, capitalize=False, **kwargs
+):
+    """
+    Usage: $your() or $your(key)
+
+    Replaces with your for the caller of the string, with the display_name +'s
+    of the caller for others.
+
+    Keyword Args:
+        caller (Object): The 'your' in the string. This is used unless another
+            your-key is passed to the callable in combination with `mapping`.
+        receiver (Object): The recipient of the string.
+        mapping (dict, optional): This is a mapping `{key:Object, ...}` and is
+            used to find which object `$you(key)` refers to. If not given, the
+            `caller` kwarg is used.
+        capitalize (bool): Passed by the You helper, to capitalize you.
+
+    Returns:
+        str: The parsed string.
+
+    Raises:
+        ParsingError: If `caller` and `receiver` were not supplied.
+
+    Notes:
+        The kwargs should be passed the to parser directly.
+
+    Examples:
+        This can be used by the say or emote hooks to pass actor stance
+        strings.
+
+        - `$your() pet jumps at $you(tommy).`
+
+        The caller-object will see "Your pet jumps Tommy."
+        Tommy will see "CharName's pet jumps at you."
+        Others will see "CharName's pet jumps at Tommy."
+
+    """
+    if args and mapping:
+        # this would mean a $your(key) form
+        try:
+            caller = mapping.get(args[0])
+        except KeyError:
+            pass
```

Comment:
When would this ever happen? `mapping.get()` returns `None` if unset, and `args[0]` should never fail since we check for `args` above (and would return `IndexError` anyway).

Reply:
That isn't true, actually. .get() will raise a KeyError when given a single argument and will return a default value if given a second. For this to return none if args[0] doesn't exist, it would need to be `.get(args[0], None)`

I copied this exactly from the normal $you(). Do you wanna rewrite both?

Reply:
Let's do that. I could do it, but a unit test need to be added to this PR anyway, so if you could fix this as well, that'd be great!